src/test/java/webserver/request/HttpRequestTest.java

```diff
@@ -55,4 +69,35 @@ void request_POST() throws IOException {
                 () -> assertThat(request.getBody("userId").get()).isEqualTo("javajigi")
         );
     }
+
+    @Test
+    void request_resttemplate() {
+        RestTemplate restTemplate = new RestTemplate();
+        ResponseEntity<String> response = restTemplate.getForEntity("http://localhost:8080", String.class);
+        assertThat(response.getStatusCode()).isEqualTo(HttpStatus.OK);
+    }
+
+    @DisplayName("설정된 쓰레드풀 보다 많은 파일 요청에도 정상 동작")
+    @Test
+    void request_index_html_at_the_same_time() throws InterruptedException {
+        ExecutorService executorService = Executors.newFixedThreadPool(100);
+        CountDownLatch latch = new CountDownLatch(100);
+        RestTemplate restTemplate = new RestTemplate();
+        AtomicInteger counter = new AtomicInteger(0);
+        StopWatch sw = new StopWatch();
+        sw.start();
+
+        for (int i = 0; i < 100; i++) {
+            executorService.execute(() -> {
+                latch.countDown();
+                int index = counter.addAndGet(1);
+                restTemplate.getForEntity("http://localhost:8080", String.class);
+                logger.info("Thread '{}'", index);
+            });
+        }
+        sw.stop();
+        executorService.shutdown();
+        latch.await(100, TimeUnit.MILLISECONDS);
+        logger.info("Total Elapsed: '{}'", sw.getTotalTimeMillis());
```

Comment:
```suggestion
        final int threadPoolCount = 100;
        final int requestCount = 200;

        ExecutorService executorService = Executors.newFixedThreadPool(threadPoolCount);
        CountDownLatch latch = new CountDownLatch(requestCount);

        StopWatch stopWatch = new StopWatch();
        stopWatch.start();

        for (int i = 0; i < requestCount; i++) {
            executorService.execute(() -> {
                RestTemplate restTemplate = new RestTemplate();
                restTemplate.getForEntity("http://localhost:8080/index.html", String.class);
            });
        }
        boolean await = latch.await(1000, TimeUnit.MILLISECONDS);
        stopWatch.stop();

        assertThat(latch.getCount()).isZero();
        assertThat(await).isTrue();

        logger.info("Total Elapsed: '{}'", stopWatch.getTotalTimeMillis());
```

이렇게 만들어보았으면 더 좋았을 것 같아요.
지금도 잘 만들어주셨어요. 👍